bgc_derivation/derivations/Derivation.m

```diff
@@ -0,0 +1,107 @@
+classdef (Abstract) Derivation
+    % Abstract class providing common functionality for derived parameters.
+    
+    % title - s Derivation vr - 1.0 author - bodc/matcaz date - 20221108
+    
+    properties
+        variables;  % Variables extracted from source data
+        default_glob;  % Default glob pattern used for selecting data files
+    end
+    properties (Abstract, Constant = true)
+        equation;  % Underlying equation function
+    end
+    methods
+        function self = Derivation(variables)
+            % Constructor for Derivation objects. Accepts a pre-configured
+            % Map of variables or can use add_ methods to ingest data.
+            %
+            % derivation = Derivation([variables])
+            % where
+            %  variables is a pre-populated name-value Map of variables.
+            %
+            
+            self.default_glob = '**/*.nc';
+            
+            if nargin == 1
+                self.variables = variables;
+            else
+                self.variables = containers.Map;
+            end
+        end
+        function add_data(self, name, data)
+            % Add a single data variable by name.
+            %
+            % derivation.add_data(name, data)
+            % where
+            %  name is the name of the supplied variable.
+            %  data is the underlying array for the supplied variable.
+            %
+            
+            self.variables(name) = data;
```

Comment:
Can this be used to save the coeff that is received from  'stcoeff = getPredeploymentCoefficients(metafilepath);'

Reply:
I think we've resolved this with our external conversation.